### ISSUE TITLE
Safari extension icon automatically moves location after an update

### ISSUE THREAD

**Issue by technima** [bitwarden/browser#2224]
### Steps To Reproduce

1. Install Bitwarden and enable the Safari extension
2. Right-click on the toolbar in Safari
3. Click "Customize Toolbar..."
4. Click and drag the Bitwarden icon to the right of the URL field and tabs list (make sure to use the "Compact" Tab layout)
5. Click done
6. Make sure Safari is open
7. Update to the December 2021 release of the app through the Mac App Store
8. Open the Bitwarden app
9. You get taken to this page (https://bitwarden.com/browser-start/), which signifies that the extension was updated
10. The extension icon gets automatically moved to the left of the URL field and tabs list

### Expected Result

The extension icon should not automatically change its location on the Safari toolbar after an update is made to the extension. The icon should stay where the user put it on the toolbar, even across updates.

### Actual Result

The extension icon automatically changes its location after an update is made to the extension. It gets moved to the immediate left of the tabs and URL fields.

### Screenshots or Videos

- I have the "Compact" Tab layout set in Safari' preferences.
- Safari is set as my default browser.

### Operating System

macOS

### Operating System Version

12.0.1

### Web Browser

Safari

### Browser Version

15.1

### Build Version

1.55.0

**Comment by Hinton** [comment]
@technima Were you using the Mac App Store version before? And was it recently updated?

**Comment by technima** [comment]
Yes, I was using the Mac App Store version before (version 1.29.1). I updated through the Mac App Store's automated update mechanism.

**Comment by technima** [comment]
@Hinton It just happened again on macOS 12.2.1 and Safari 15.3 when I upgraded to version 1.56.5 of the extension that's bundled with the Mac App Store Bitwarden app.

**Comment by atjbramley** [comment]
I can provide a video of this behaviour:

### Screenshots or Videos

[  See attached.](https://user-images.githubusercontent.com/71517198/192548278-ad377634-43d3-4e44-9320-b398a517180f.mp4)

### Operating System

macOS

### Operating System Version

12.6

### Web Browser

Safari

### Browser Version

16

### Build Version

2022.8.0

**Comment by technima** [comment]
@atjbramley That's exactly what I'm seeing on my end. And the issue has persisted on the latest version of the extension, 2022.8.0

**Comment by radon-00** [comment]
The bug still exists with Bitwarden Version 2022.11.0 and newest MacOS 13.1 with Safari Version 16.2 (18614.3.7.1.5).

**Comment by radon-00** [comment]
The bug still exists with the new version 2023.1.1 and same MacOS / Safari version as above.

**Comment by Hinton** [comment]
Unfortunately we don't have a way to specify the location of the browser extension. Safari decides the initial position and the user is able to modify it afterwards. Since this seems to happen on either update or when Safari changes the permissions of the extension it seems that Safari in the process of reloading the extension also forgets the previous location.

I'm unfortunately unable to replicate the behavior on my machine, tried toggling the permissions of the extension after moving the extension as well as updating it. Would appreciate if an affected user could file a bug either on the webkit bug tracker detailing their steps to reproduce it, or through the Mac Feedback Assistance.

**Comment by noahmaranesi** [comment]
**This is a macOS and/or Safari bug.**

I came across this thread from a Google search because I'm sick of extensions relocating after the app gets updated.

**Comment by closebot-bw** [comment]
⚠️ **Stale Issue Notice**

This issue has been automatically marked as stale due to inactivity. It will be closed in **2 weeks** (October 15, 2025) if no further activity occurs.

If this issue is still relevant and you would like to keep it open, please:
- Comment on this issue to show continued interest
- Provide any additional information or updates
- Confirm that the issue still exists in the latest version

Thank you for your contribution to this project! 🙏

**Comment by closebot-bw** [comment]
🔒 **Issue Closed Due to Inactivity**

This issue has been automatically closed due to lack of activity for an extended period. We periodically review and close inactive issues to help maintain our issue tracker and focus on current priorities.

**If this issue is still relevant:**
- Please create a new issue with updated information
- Include steps to reproduce the problem if it's a bug report
- Mention if this issue still occurs in the latest version

Thank you for your contribution to this project. Your feedback helps us improve! 🙏
